operator/operator.php

```diff
@@ -80,6 +105,16 @@ public function get_cookie_categories()
 		// TODO
 		// 1.gets categories from the cookie (will need to be un-encoded by json decode and htmlspecialchars decode)
 		// 2.return categories or an empty array
+
+		$cookie_data = '';
+		$cookie_categories = array();
+		if ($this->request->is_set($this->config['cookie_name'] . '_ccat', \phpbb\request\request_interface::COOKIE))
+		{
+			$cookie_data = htmlspecialchars($this->request->variable($this->config['cookie_name'] . '_ccat', '', true, \phpbb\request\request_interface::COOKIE));
+			$cookie_categories = (array) json_decode($cookie_data);
```

Comment:
json_decode needs a true for second parameter so it returns and array instead of an object